session_items.py

```diff
@@ -67,3 +65,35 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+#loop through exisiting items!!!!
+
+def delete_item(item_id):
+
+    existing_items = get_items()
+    session['items'] = [ items for items in existing_items if int(items.get('id')) != int(item_id) ]
```

Comment:
Based on your experimental remove_from_list.py file I assume you had a little difficulty with this.

The surprising thing with session['items'] is that if you modify the list in place, you need to let it know to update the user's cookie, by doing `session.modified = True`. It won't notice automatically if you do this for example: `session['items'].remove(get_item(item_id))`